neuralpredictors/layers/readouts/gaussian.py

```diff
@@ -380,6 +389,23 @@ def feature_l1(self, reduction="sum", average=None):
         else:
             return 0
 
+    def adaptive_l1_lognorm(self, reduction="sum", average=None):
+        if self._original_features:
+            features = self.adaptive.abs() * self.features
+            return self.apply_reduction(features.abs(), reduction=reduction, average=average)
+        else:
+            return 0
+
+    def regularizer(self, reduction="sum", average=None):
+        if self.regularizer_type == "l1":
+            return self.feature_l1(reduction=reduction, average=average) * self.feature_reg_weight
+        else:
```

Comment:
Isn't it redundant because we already check `self.regularizer_type` to be valid in [lines 291-295](https://github.com/sinzlab/neuralpredictors/blob/80279a0f0c773c73c246981bbb493e05f7487328/neuralpredictors/layers/readouts/gaussian.py#L291-L295) ?

Reply:
computationally yes, it improves code readability though imo. But no strong opinion here. In my own code, I sometimes also just denote the other option as a comment, e.g.

```suggestion
        else:  # self.regularizer_type == "adaptive_log_norm"
```

I would leave this decision up to you